codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/JsonRpcProtocolGenerator.java

```diff
@@ -163,6 +164,12 @@ protected void deserializeError(GenerationContext context, StructureShape shape)
         writer.write("");
         handleDecodeError(writer);
         writer.write("errorBody.Seek(0, io.SeekStart)");
+        if (context.getService().hasTrait(AwsQueryCompatibleTrait.class)) {
```

Comment:
Is it possible to move this to an overridden `deserializeError()` method in `AwsJsonRpc1_0.java`?

This will keep all of the changes to 1 file, and in the off-chance another JSON RPC protocol somehow triggers this code path, the generated Go code will break since `getAwsQueryErrorCode()` is only generated in `AwsJsonRpc1_0.java`.

Duplication isn't much of an issue.